catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/XmlModelBuilder.java

```diff
@@ -142,35 +140,35 @@ public XmlModelBuilder beginBinaryComparisonType(String operator) {
     verifyResultNotYetRetrieved();
     verifyTerminalNodeNotInProgress();
     MultiNodeReducer comparisonMapping = TERMINAL_OPS.get(operator);
-    if (comparisonMapping == null) {
-      throw new IllegalArgumentException(
-          "Cannot find mapping for binary comparison operator: " + operator);
-    }
+    validateOperatorMapping(
+        comparisonMapping, "Cannot find mapping for binary comparison operator: " + operator);
     supplierInProgress = new TerminalNodeSupplier(comparisonMapping);
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginPropertyIsLikeType(String operator, boolean matchCase) {
+  public FlatFilterBuilder beginPropertyIsLikeType(String operator) {
     verifyResultNotYetRetrieved();
     verifyTerminalNodeNotInProgress();
     MultiNodeReducer comparisonMapping = TERMINAL_OPS.get(operator);
-    if (comparisonMapping == null) {
-      throw new IllegalArgumentException("Cannot find mapping for like operator: " + operator);
-    }
+    validateOperatorMapping(
+        comparisonMapping, "Cannot find mapping for like operator: " + operator);
     supplierInProgress = new TerminalNodeSupplier(comparisonMapping);
     return this;
   }
 
+  @Override
+  public FlatFilterBuilder beginPropertyIsILikeType(String operator) {
+    return beginPropertyIsLikeType(operator);
```

Comment:
:question: I don't see the part of this code that differentiates between `ILIKE` and `LIKE` comparisons. Are they meant to be equivalent? If so, why? Isn't `ILIKE` case-insensitive while `LIKE` is case-sensitive?

Reply:
There are two directions in play here, JSON to XML and vice versa. At best, we can simply move the "kludgy-ness" around the codebase without removing it completely (that will come later when Geotools gets upgraded, hopefully). 

Normally Filter 2.0 only has a `PropertyIsLikeType` which itself has a boolean attribute for `matchCase`. Since we are mapping to the UI's model, and it distinguishes case sensitivity as two different operations (the CQL `LIKE` and `ILIKE` operators) we decided it best to be declarative until subsequent improvements can be made.

Reply:
So if I'm understanding correctly, Filter 2.0 can really only do `LIKE`, but we have to support `ILIKE`, so this makes `ILIKE` just work like `LIKE`?

Reply:
Glen helped explain in person that the operator string "LIKE" or "ILIKE"
 is passed into this method; then, at [this line in `beginPropertyIsLikeType`](https://github.com/codice/ddf/pull/3331/files/154125c740d717194b4078b821d0c874b2c0b01c#diff-9c1e9e448375a9b21e42d9b1e5e109a2R153), the appropriate operation is retrieved using the operator as the key "LIKE" or "ILIKE" from the map `TERMINAL_OPS`. [These two lines](https://github.com/codice/ddf/pull/3331/files/154125c740d717194b4078b821d0c874b2c0b01c#diff-9c1e9e448375a9b21e42d9b1e5e109a2R63) add LIKE and ILIKE to `TERMINAL_OPS` so that either can be found with this map retrieval.

That's how this code differentiates between LIKE and ILIKE even though both are calling `beginPropertyIsLikeType`.

Consider this :pencil2:, but I think that this confusion could be avoided by factoring the body of `beginPropertyIsLikeType` into a new private method, e.g., `beginPropertyIsLikeOrILikeType`, and making `beginPropertyIsLikeType` and `beginPropertyIsILikeType` call that.

Reply:
That's a good idea. Done.